src/app/components/tables/data-table/terra-data-table.component.html

```diff
@@ -1,60 +1,57 @@
-<div class="group-function" [@collapsedState]="getCollapsedState">
-    <div class="group-function-head">
-        <terra-button inputIcon="icon-execute"
-                      [inputIsDisabled]="inputGroupFunctionExecuteButtonIsDisabled"
-                      (outputClicked)="onGroupFunctionExecuteButtonClicked($event)">
-        </terra-button>
-        <span>{{'terraDataTable.groupFunction' | translate:lang}}</span>
-        <span class="icon-collapse_down" [class.rotated]="inputCollapsed" *ngIf="inputIsCollapsable"></span>
-    </div>
-    <div class="group-function-body">
-        <ng-content></ng-content>
-    </div>
-</div>
+<ng-content select="terra-group-function"></ng-content>
 
 <terra-pager *ngIf="inputHasPager && rowList && rowList.length > 0"
-             [inputPagingSize]="pagingSize"
-             [inputDefaultPagingSize]="defaultPagingSize"
-             [inputPagingData]="pagingData"
-             [inputRequestPending]="requestPending"
+             [inputPagingSize]="inputService?.pagingSizes"
+             [inputDefaultPagingSize]="inputService?.defaultPagingSize"
+             [inputPagingData]="inputService?.pagingData"
+             [inputRequestPending]="inputService?.requestPending"
              (outputDoPaging)="doPaging($event)">
 </terra-pager>
 
 <div class="table-responsive">
-    <context-menu></context-menu>
-    <table class="table data-table">
+    <context-menu [links]="inputContextMenu"></context-menu>
+    <table class="table data-table" *ngIf="isTableDataAvailable">
         <thead>
         <tr>
-            <th *ngIf="inputHasCheckboxes && rowList.length" width="25px">
-                <terra-checkbox #viewChildHeaderCheckbox
-                                (change)="onHeaderCheckboxChange(viewChildHeaderCheckbox.value)"
-                                [value]="isHeaderCheckboxChecked">
+            <th *ngIf="inputHasCheckboxes" width="25px">
+                <terra-checkbox (change)="onHeaderCheckboxChange()"
+                                (click)="$event.stopPropagation()"
+                                [value]="headerCheckbox.checked"
+                                [isIndeterminate]="headerCheckbox.isIndeterminate">
                 </terra-checkbox>
             </th>
-            <ng-template ngFor let-header [ngForOf]="headerList">
-                <th *ngIf="!header.isHidden && rowList && rowList.length > 0"
+            <ng-template ngFor let-header [ngForOf]="inputHeaderList">
+                <th *ngIf="!header.isHidden"
                     width="{{header.width}}px"
```

Comment:
Wertegenerierung von width in Funktion auslagern

Reply:
ungern.. Ein Funktionsaufruf bei jedem Changedetection-Zyklus scheint mir weniger performant zu sein als diese Variante